app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/item/component/ItemHeader.kt

```diff
@@ -0,0 +1,399 @@
+    isExpanded: Boolean,
+    iconTestTag: String? = null,
+    textFieldTestTag: String? = null,
+    onExpandClick: () -> Unit,
+) {
+    item {
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            modifier = Modifier
+                .fillMaxWidth()
+                .standardHorizontalMargin()
+                .defaultMinSize(minHeight = 60.dp)
+                .cardStyle(
+                    cardStyle = CardStyle.Top(),
+                    paddingVertical = 0.dp,
+                )
+                .padding(start = 16.dp),
+        ) {
+            ItemHeaderIcon(
+                iconData = iconData,
+                testTag = iconTestTag,
+                modifier = Modifier.size(36.dp),
+            )
+            BitwardenTextField(
+                label = null,
+                value = value,
+                onValueChange = { },
+                readOnly = true,
+                singleLine = false,
+                actions = {
+                    Icon(
+                        painter = painterResource(
+                            id = if (isFavorite) {
+                                R.drawable.ic_favorite_full
+                            } else {
+                                R.drawable.ic_favorite_empty
+                            },
+                        ),
+                        contentDescription = stringResource(
+                            id = if (isFavorite) R.string.favorite else R.string.unfavorite,
+                        ),
+                        modifier = Modifier.padding(all = 12.dp),
+                    )
+                },
+                textFieldTestTag = textFieldTestTag,
+                cardStyle = null,
+                textStyle = BitwardenTheme.typography.titleMedium,
+            )
+        }
+    }
+
+    if (relatedLocations.isEmpty()) {
+        item {
+            ItemLocationListItem(
+                vectorPainter = rememberVectorPainter(R.drawable.ic_folder),
+                text = stringResource(R.string.no_folder),
+                iconTestTag = "NoFolderIcon",
+                modifier = Modifier
+                    .standardHorizontalMargin()
+                    .fillMaxWidth()
+                    .cardStyle(
+                        cardStyle = CardStyle.Bottom,
+                        paddingVertical = 0.dp,
+                    ),
+            )
+        }
+        return
+    }
+
+    items(
+        key = { "locations_$it" },
+        items = relatedLocations.take(EXPANDABLE_THRESHOLD),
+    ) {
+        ItemLocationListItem(
+            vectorPainter = rememberVectorPainter(it.icon),
+            iconTestTag = "ItemLocationIcon",
+            text = it.name,
+            modifier = Modifier
+                .fillMaxWidth()
+                .standardHorizontalMargin()
+                .animateItem()
+                .cardStyle(
+                    cardStyle = if (relatedLocations.size > EXPANDABLE_THRESHOLD) {
```

Comment:
I think we only want this to be `CardStyle.Bottom` if it's the last item to be shown.